tests/StorageApiTestCase.php

```diff
@@ -673,7 +685,140 @@ function (self $that, array $config, $useExistingBranch = false) {
+                }
+            }
+        ];
+
+        yield 'devBranch' => [
+            function (self $that, array $config = [], $useExistingBranch = false) {
+                $branch = $this->createOrReuseDevBranch($that, $useExistingBranch);
+
+                if ($config) {
+                    return $this->getBranchAwareClient($branch['id'], $config);
+                } else {
+                    return $this->getBranchAwareDefaultClient($branch['id']);
+                }
+            }
+        ];
+    }
+
+    protected function getDefaultBranchId(self $that)
+    {
+        $devBranch = new \Keboola\StorageApi\DevBranches($that->_client);
+        $branchesList = $devBranch->listBranches();
+        foreach ($branchesList as $branch) {
+            if ($branch['isDefault'] === true) {
+                return $branch['id'];
+            }
+        }
+
+        return null;
+    }
+
+    public function cleanupConfigurations()
+    {
+        $components = new Components($this->_client);
+        foreach ($components->listComponents() as $component) {
+            foreach ($component['configurations'] as $configuration) {
+                $components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+
+        // erase all deleted configurations
+        foreach ($components->listComponents((new ListComponentsOptions())->setIsDeleted(true)) as $component) {
+            foreach ($component['configurations'] as $configuration) {
+                $components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+    }
+
+    /**
+     * @param string $eventName
+     * @return array
+     */
+    protected function listEvents(Client $client, $eventName, $expectedObjectId = null)
+    {
+        return $this->retry(function () use ($client, $expectedObjectId) {
+            $tokenEvents = $client->listEvents([
+                'sinceId' => $this->lastEventId,
+                'limit' => 1,
+                'q' => sprintf('token.id:%s', $this->tokenId),
+            ]);
+
+            if ($expectedObjectId === null) {
+                return $tokenEvents;
+            }
+
+            return array_filter($tokenEvents, function ($event) use ($expectedObjectId) {
+                return $event['objectId'] === $expectedObjectId;
+            });
+        }, 10, $eventName);
+    }
+
+    /**
+     * @param callable $apiCall
+     * @param int $retries
+     * @param string $eventName
+     * @return array
+     */
+    protected function retry($apiCall, $retries, $eventName)
+    {
+        $events = [];
+        while ($retries > 0) {
+            $events = $apiCall();
+            if (empty($events) || $events[0]['event'] !== $eventName) {
+                $retries--;
+                usleep(250 * 1000);
+            } else {
+                break;
+            }
+        }
+        return $events;
+    }
```

Comment:
Vlastní implementace retry, nemá testy :(

Reply:
Já vím :( Ale jde mi o to, abysme to nějak řešili. Teď už se to nepoužívá na tom "jediném místě, kde se to napsalo jako MVP", ale už je to na dvou místech. A za čas jsme v tom, že se to používá standardně a prostě tím normalizujeme to, že je OK psát někde takovýhle random kód :( 

https://keboola.atlassian.net/browse/KBC-1937

Reply:
jo tak som to zle pochopil, diky za zalozenie story